src/anndata/_io/specs/registry.py

```diff
@@ -34,6 +40,54 @@
     LazyDataStructures = DaskArray | Dataset2D | CategoricalArray | MaskedArray
 
 
+def is_sparse_like(x):
+    try:
+        return sp.issparse(x)
+    except AttributeError:
+        return False
+
+
+def to_numpy_if_array_api(x):
```

Comment:
You probably want to check `device` first to weed out gpu arrays https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.device.html and/or bring them back to the CPU